pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -287,10 +287,12 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 					reqLogger.Error(err, "failed to create external Service", "Service", ser)
 					return reconcile.Result{}, err
 				}
-				err = r.client.Update(context.TODO(), infinispan)
-				if err != nil {
-					reqLogger.Error(err, "Failed to update Infinispan with Service", "Service", ser)
-					return reconcile.Result{}, err
+				if len(externalService.Spec.Ports) > 0 {
+					infinispan.Spec.Expose.NodePort = externalService.Spec.Ports[0].NodePort
+					err = r.client.Update(context.TODO(), infinispan)
```

Comment:
@rigazilla, do we need to get actual externalService record after create or it will return to the variable back?
As for as I remember, some of the external services, like LoadBalancer are created asynchronously as contains of set of the steps on the underlying platform, i.e. AWS, etc.